calico_rkt/calico_rkt.py

```diff
@@ -1,91 +1,249 @@
+        delete(
+            container_id=env['CNI_CONTAINERID']
+            )
+
+def create(container_id):
+    """"Handle rkt pod-create event."""
+    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
+    _datastore_client = IPAMClient()
 
-    # Convert input data to JSON object
     try:
-        stdin_json = json.loads(stdin_raw_data)
-    except ValueError as e:
-        quit_with_error(str(e))
-
-    # Extract command
+        endpoint, ip = _create_calico_endpoint(container_id=container_id,
+                                            netns_path=netns_path,
+                                            client=_datastore_client)
+
+        _create_profile(endpoint=endpoint, 
+                        profile_name=input_json['name'], 
+                        ip=ip,
+                        client=_datastore_client)
+    except CalledProcessError as e:
+        print_stderr('Error code %d creating pod networking: %s\n%s' % (
+            e.returncode, e.output, e))
+        sys.exit(1)
+    print_stderr('Finished Creating pod %s' % container_id)    
+
+def delete(container_id):
+    """Cleanup after a pod."""
+    print_stderr('Deleting pod %s' % container_id, file=sys.stderr)
+
+    _datastore_client = IPAMClient()
+
+    # Remove the profile for the workload.
+    _container_remove(hostname=HOSTNAME,
+                        orchestrator_id=ORCHESTRATOR_ID,
+                        container_id=container_id,
+                        client=_datastore_client)
+
+    # Delete profile
     try:
-        command = stdin_json['command']
-    except KeyError:
-        quit_with_error("Missing command")
-
-    # Extract args
+        _datastore_client.remove_profile(input_json['name'])
+    except:
+        print_stderr("Cannot remove profile %s; Profile cannot be found." % container_id)
+
+def _create_calico_endpoint(container_id, netns_path, client):
+    """Configure the Calico interface for a pod."""
+    print_stderr('Configuring Calico networking.', file=sys.stderr)
+
+    interface = env['CNI_IFNAME']
+
+    endpoint, ip = _container_add(hostname=HOSTNAME,
+                                orchestrator_id=ORCHESTRATOR_ID,
+                                container_id=container_id,
+                                netns_path=netns_path,
+                                interface=interface,
+                                client=client)
+    print_stderr('Finished configuring network interface', file=sys.stderr)
+    return endpoint, ip
+
+def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, client):
+    """
+    Add a container to Calico networking with the given IP
+    Return Endpoint object
+    """
     try:
-        args = stdin_json['args']
+        _ = client.get_endpoint(
+            hostname=hostname,
+            orchestrator_id=orchestrator_id,
+            workload_id=container_id
+        )
```

Comment:
This alignment isn't quite right, but I don't think the alignment should be done like this anyway.

I think a more *pythonic* way of writing this is to have the first param on the same line as the call (like line 98 with `_container_add`, then aligning the other params with the first:
```
func_call(param_1,
          param_2,
          param_3)
```